validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockFactory.java

```diff
@@ -296,4 +307,22 @@ private ExecutionPayload validateExecutionPayload(
 
     return executionPayload;
   }
+
+  private static class PreparePayloadReference {
```

Comment:
I was thinking that to do a complete validation of the payload coming from the execution client, we should validate all params (except the `feeRecipient` which is just a suggestion).

This means that we should cache `executionParentHash`, `timestamp` and `random` and then validate them in `validateExecutionPayload` which then except the entire `PreparePayloadReference` class instance as a param.

This way we are checking all values the consensus side is responsible for.

I'm wandering if before publishing the whole block it will be in any case validated by `BlockProcessorMerge::processBlock` and then `BlockProcessorMerge::processExecutionPayload`

Reply:
> I'm wandering if before publishing the whole block it will be in any case validated by BlockProcessorMerge::processBlock and then BlockProcessorMerge::processExecutionPayload

looking in the code flow I don't see those kind of validation to run after block production.
So, on one hand I'm incline to say that we should validate payload fields that fall under consensus responsibility. But, on the other hand, if execution client goes crazy and start including invalid txs, as consensus we accept the payload but at the and the entire block will be rejected by the network... and since there are no additional penalties by producing an invalid block compared to non publishing it at all, maybe those kind of checks are not that mandatory..

Reply:
The execution engine is inherently trusted - we depend on it to tell us if the payload executes correctly already.  So if it's easy to add sanity checks that we get back what we should that makes sense and will let us track down the source of the problem more easily but if it adds any complexity I'd leave it out.

Reply:
Yep, I tend to agree with @ajsutton and leave it for now. 

> I'm wandering if before publishing the whole block it will be in any case validated by BlockProcessorMerge::processBlock and then BlockProcessorMerge::processExecutionPayload

I don't think they are validated that way but might be a handy debug feature